### PR TITLE
Redefine crossScalaVersions, because it's Def.derive..

### LINKED ISSUE #3495: Switching Scala version with ++ modifies crossScalaVersions (1.0 regression)
## steps

1. Create a project with more than one `crossScalaVersions`
1. Open SBT shell and inspect `crossScalaVersions` - it should list versions specified in `build.sbt`
1. Switch Scala version using `++`, e.g. `++2.11.11`
1. Inspect `crossScalaVersions` again - it should list the same versions as before, but it lists only the one consistent with current `scalaVersion`

```
scala-commons$ sbt
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /home/ghik/idea-workspace/scala-commons/project
[info] Loading settings from build.sbt,version.sbt ...
[info] Set current project to commons (in build file:/home/ghik/idea-workspace/scala-commons/)
[info] sbt server started at 127.0.0.1:4456
sbt:commons> commons-core/scalaVersion
[info] 2.12.3
sbt:commons> commons-core/crossScalaVersions
[info] * 2.11.11
[info] * 2.12.3
sbt:commons> ++2.11.11
[info] Setting Scala version to 2.11.11 on 13 projects.
[info] Reapplying settings...
[info] Set current project to commons (in build file:/home/ghik/idea-workspace/scala-commons/)
sbt:commons> commons-core/scalaVersion
[info] 2.11.11
sbt:commons> commons-core/crossScalaVersions
[info] * 2.11.11
sbt:commons>
```

## problem

`crossScalaVersions` is filtered when switching Scala version with `++`

## expectation

`crossScalaVersions` should not be modified when switching Scala version

## notes

This bug has a very nasty consequence - all cross-building commands starting with `+` (e.g. `+compile`) no longer work because after `crossScalaVersions` had been filtered it is impossible to switch again to any other Scala version using `++`.

sbt version: 1.0.1

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #3495

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=pull/3526/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

File: main/src/main/scala/sbt/Cross.scala [hunk 1]
```diff
@@ -274,12 +274,14 @@ object Cross {
...
-    val projects: Seq[ResolvedReference] = {
+    val projects: Seq[(ResolvedReference, Seq[ScalaVersion])] = {
       val projectScalaVersions =
         structure.allProjectRefs.map(proj => proj -> crossVersions(extracted, proj))
       if (switch.version.force) {
         logSwitchInfo(projectScalaVersions, Nil)
-        structure.allProjectRefs ++ structure.units.keys.map(BuildRef.apply)
+        projectScalaVersions ++ structure.units.keys
+          .map(BuildRef.apply)
+          .map(proj => proj -> crossVersions(extracted, proj))
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Cross.scala:284]: So you're getting the original here and re-applying it at project-level later?

**Review by eed3si9n — CHANGES_REQUESTED** [review]
overall LGTM
release note plz

File: main/src/main/scala/sbt/Cross.scala (hunk 1 above)
Comment (dwijnand) [reply main/src/main/scala/sbt/Cross.scala:284]: This, specifically, is the "force" branch, but more generally yes that's the workaround.

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=pull/3526/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**